lib/cloudware/aws.rb

```diff
@@ -119,32 +121,31 @@ def machines
-                          if test.parameters[0]
-                            @state = 'failed'
-                            @extip = 'N/A'
-                            @instance_id = 'N/A'
-                            test.parameters.each_with_index do |val, index|
-                              @type = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmType'
-                              @domain = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareDomain'
-                              @id = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareId'
-                              @role = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmRole'
-                              @prvip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'prvIp'
-                              @mgtip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'mgtIp'
-                              @flavour = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmFlavour'
-                            end
-                            @name = test.stack_name.gsub("#{@domain}-", '')
-                            log.info("[#{self.class}] Detected machine #{@name} in domain #{@domain}")
-                            @machines.merge!("#{@domain}-#{@name}" => {
-                                               name: @name, domain: @domain, state: @state,
-                                               id: @id, type: @type, role: @role, mgt_ip: @mgtip,
-                                               prv_ip: @prvip, ext_ip: @extip, provider: 'aws', instance_id: @instance_id,
-                                               flavour: @flavour
-                                             })
+                          next unless test.stack_status.include?('ROLLBACK')
+                          next unless test.parameters[0]
+                          @state = 'failed'
+                          @extip = 'N/A'
+                          @instance_id = 'N/A'
+                          test.parameters.each_with_index do |val, _index|
+                            @type = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmType'
+                            @domain = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareDomain'
+                            @id = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'cloudwareId'
+                            @role = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmRole'
+                            @prvip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'prvIp'
+                            @mgtip = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'mgtIp'
+                            @flavour = val.to_h[:parameter_value] if val.to_h[:parameter_key] == 'vmFlavour'
                           end
+                          @name = test.stack_name.gsub("#{@domain}-", '')
```

Comment:
I believe this block (up to line 148) needs to be within the loop `@cfn.describe_stacks.stacks.each do |test|`, this is looping through the different stacks and adding their details to the `@machines` list. With it being moved to here it's now part of the region loop which I do not believe will work.

Reply:
I am unsure what you mean. from the looks of it, lines 141-148 are within the `@cfn.describe_stacks.stacks.each do |test|` block.

The `end` on line 149 closes the block (starting on line 126). The `end` on line 140 closes the nested block starting on line 132.

This was only a rubocop change, it was correcting some indenting issues

Reply:
I am unsure what you mean. from the looks of it, lines 141-148 are within the `@cfn.describe_stacks.stacks.each do |test| block`.

The end on line 149 closes the block (starting on line 126). The end on line 140 closes the nested block starting on line 132.

This was only a rubocop change, it was correcting some indenting issues

Reply:
Too right. My apologies, didn't notice the other block. LGTM.